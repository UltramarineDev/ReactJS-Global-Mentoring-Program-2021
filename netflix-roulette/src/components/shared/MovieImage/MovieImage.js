import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import styles from './MovieImage.module.scss';

const MovieImage = ({ imageUrl }) => {
  const [hasImageError, setImageError] = useState(false);

  return (
    <div className={hasImageError ? styles.noImageWrapper : styles.image}>
      {hasImageError
        ? <div className={styles.noImage}><FontAwesomeIcon icon="image" /></div>
        : <img src={imageUrl} alt="Movie image" onError={() => setImageError(true)} />}
    </div>
  );
};

MovieImage.propTypes = {
  imageUrl: PropTypes.string.isRequired,
};

export default MovieImage;
