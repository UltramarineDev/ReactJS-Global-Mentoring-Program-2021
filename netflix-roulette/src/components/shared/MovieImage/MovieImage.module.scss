.image {
  margin-bottom: 16px;
  height: 450px;
  background-color: 232323;
  width: 300px;
}

.noImageWrapper {
  @extend .image;

  border: solid thin;
}

.image img {
  min-width: 300px;
  width: 300px;
  height: 450px;
}

.noImage {
  font-size: 80px;
  top: 30%;
  position: absolute;
  font-weight: 23px;
  left: 30%;
}