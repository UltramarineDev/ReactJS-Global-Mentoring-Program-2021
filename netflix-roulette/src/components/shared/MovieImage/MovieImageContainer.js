import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import MovieImage from './MovieImage';

class MovieImageContainer extends PureComponent {
  static propTypes = {
    imageUrl: PropTypes.string.isRequired,
  };

   state = {
    hasImageError: false,
  }

  handleImageError = () => this.setState({ hasImageError: true });

  render() {
    const { imageUrl } = this.props;
    const { hasImageError } = this.state;

    return (
      <MovieImage
        onImageError={this.handleImageError}
        hasImageError={hasImageError}
        imageUrl={imageUrl}
      />
    );
  }
};

export default MovieImageContainer;
