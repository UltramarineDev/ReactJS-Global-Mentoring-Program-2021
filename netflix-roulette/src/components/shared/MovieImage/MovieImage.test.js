import React from 'react';
import { render, cleanup } from '@testing-library/react';

import MovieImage from './MovieImage';

describe('MovieCard tests', () => {
  afterEach(() => {
    cleanup();
  });
  it('renders MovieImage', () => {
    const { asFragment } = render(<MovieImage />);
    expect(asFragment(<MovieImage />)).toMatchSnapshot();
  });
});
