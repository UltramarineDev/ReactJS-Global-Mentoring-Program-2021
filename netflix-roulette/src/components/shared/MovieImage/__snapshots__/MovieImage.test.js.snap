// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard tests renders MovieImage 1`] = `
<DocumentFragment>
  <div
    class="noImageWrapper"
  />
</DocumentFragment>
`;
